### PR TITLE
Add support for native Mysql blob types and fallback for binary

### PR DESCRIPTION
**Pull request by liqueurdetoile**
Hi,

This PR came from discussion here : https://github.com/cakephp/migrations/issues/359

There's no breaking change.

The extended behavior is the ability with `MysqlAdapter` to :

- use `binary` or `varbinary` type with any length and get the best fitted column type
- directly use `tinyblob`, `mediumblob` or `longblob` with no limit to get this column type and right length settled

In case of conflict between required type and limit, the limit is used to select the best fitted column type.

Aside this, I've also fix some tests that were relying on hard coded db name and CR/LF issues for string comparisons

Hope it can be useful and that I've not mistaken any guideline :smile:

### CODE REVIEW

File: src/Phinx/Db/Adapter/MysqlAdapter.php [hunk 1]
```diff
@@ -922,28 +925,57 @@ public function getSqlType($type, $limit = null)
 
                 return ['name' => 'text'];
             case static::PHINX_TYPE_BINARY:
-                return ['name' => 'binary', 'limit' => $limit ?: 255];
+                if (is_null($limit)) {
```
Comment (dereuromark) [inline src/Phinx/Db/Adapter/MysqlAdapter.php:928]: `=== null` better

Comment (liqueurdetoile) [reply src/Phinx/Db/Adapter/MysqlAdapter.php:928]: Miss that one

**Comment by dereuromark** [comment]
There seems to be a conflict.

**Comment by liqueurdetoile** [comment]
There's no conflict in my opinion. Previous commit simply replace hard coded db name with another one. There's no way to have a `phinx_testing` nor a `phinx` database settled in these tests apart config DSN config setup in `phpunit.xml`.

Im' pretty confident that my replacements should be kept as it uses DSN db name and will work in any setup.

**Comment by othercorey** [comment]
It isn’t an opinion. git just requires merging master to your branch because of edits in the same file.

**Comment by liqueurdetoile** [comment]
Not really, you've switched back to hard coded db name that will make tests fails in local dev setup if not choosing 'phinx' as db name.

**Comment by dereuromark** [comment]
> Conflicting files
tests/Phinx/Db/Adapter/MysqlAdapterTest.php

Maybe you didnt make the changes on top of latest master?
You might want to rebase.

**Comment by liqueurdetoile** [comment]
Conflict is from a PR merged after mine was submitted. I propose to leave hard-coded way :smile:

**Comment by othercorey** [comment]
The merge looks good. It should use the new config array like you have it.

File: src/Phinx/Db/Adapter/MysqlAdapter.php (hunk 1 above)
Comment (dereuromark) [reply src/Phinx/Db/Adapter/MysqlAdapter.php:928]:
```suggestion
                if ($limit === null) {
```

File: src/Phinx/Db/Adapter/MysqlAdapter.php [hunk 2]
```diff
@@ -1103,9 +1136,19 @@ public function getPhinxType($sqlTypeDef)
                 $limit = static::TEXT_LONG;
                 break;
             case 'binary':
+                if (is_null($limit)) {
```
Comment (dereuromark) [inline src/Phinx/Db/Adapter/MysqlAdapter.php:1139]: Same here

File: tests/Phinx/Db/Adapter/MysqlAdapterTest.php [hunk 3]
```diff
@@ -890,6 +892,102 @@ public function testTinyTextColumn()
...
+    /** @dataProvider varbinaryToBlobAutomaticConversionData */
+    public function testVarbinaryToBlobAutomaticConversion(?int $limit = null, string $expectedType, int $expectedLimit)
+    {
+        $table = new \Phinx\Db\Table('t', [], $this->adapter);
+        $table->addColumn('column1', 'varbinary', ['limit' => $limit])
+              ->save();
+        $columns = $table->getColumns();
+        $sqlType = $this->adapter->getSqlType($columns[1]->getType(), $columns[1]->getLimit());
+        $this->assertEquals($expectedType, $sqlType['name']);
+        $this->assertEquals($expectedLimit, $columns[1]->getLimit());
```
Comment (dereuromark) [inline tests/Phinx/Db/Adapter/MysqlAdapterTest.php:946]: Can we please use assertSame() everywhere to make sure the types are also as expected?

Comment (liqueurdetoile) [reply tests/Phinx/Db/Adapter/MysqlAdapterTest.php:946]: Sure. Should I also switch all others assertEquals ? I've simply mirrored pattern found in test file.

Comment (dereuromark) [reply tests/Phinx/Db/Adapter/MysqlAdapterTest.php:946]: If you want, sure.
We can also do that in a separate PR for other code, though.

**Comment by dereuromark** [comment]
Some errors seem left:

```
There were 4 errors:

1) Test\Phinx\Db\Adapter\MysqlAdapterTest::testCreateTableWithForeignKeys
Undefined index: table_name

/home/runner/work/phinx/phinx/tests/Phinx/Db/Adapter/MysqlAdapterTest.php:207

2) Test\Phinx\Db\Adapter\MysqlAdapterTest::testAddPrimarykey
Undefined index: constraint_name

...
```

**Comment by othercorey** [comment]
Those failures started with mysql 8.0.21 in other PRs.

**Comment by dereuromark** [comment]
Weird, master is green ( https://travis-ci.com/github/cakephp/phinx/builds/176148908 )
Maybe this PR might need another merge of master?

**Comment by liqueurdetoile** [comment]
Thx @othercorey 

I was wondering how the hell it may have been related to this PR.

**Comment by othercorey** [comment]
This only started with the mysql 8.0.21 release. You can see my draft PR showing 8.0.20 builds.

**Comment by dereuromark** [comment]
Merging then, and we figure out to solve travis in the separate PR.

**Pull request merged**